### ISSUE TITLE
Three possibly related issues with input in Awesome

### ISSUE THREAD

**Issue by maksbotan** [awesomeWM/awesome#696]
Hi.
I know this was reported many times and said to be fixed in 3.5.7/3.5.7, but it does not seem to be fixed for me for some reason.

So, here are the issues:
- wibox does not respond to mouse clicks when keyboard layout is different from english. This was announced to be fixed, but does not work for me.
- Awesome sometimes suddenly stops reacting to all mouse events, regardless of keyboard layout. Fixed by restarting all Qt5 apps. This happens _without_ plugging input devices, so different from other similar reports.
- Google chrome, again suddenly, stops reacting to keyboard, including `Mod4-Shift-C`.

I was not able to trace exact reasons of this behavior and I'm completely frustrated with this.

It'd be great if you gave me any hints about what information I should provide.

**Comment by psychon** [comment]
Replying to each point in turn:

* Where was this announced to be fixed? Some change in some xkb-related-something (was it xkbcomp?) uncovered a bug in Awesome. I still don't know/understand the details, but AFAIK this is fixed in git/master, so will be fixed in awesome 3.6. The fix is to use the XKB extension instead of relying on (broken) X11 core keyboard "stuff". The switch to XKB is complicated enough that it won't be backported to 3.5. (Someone once tried that, but didn't succeed; duplicate of #196)
* This sounds a lot like "Qt is stealing input". For debugging this, I'd enable some debug output from Qt (it was some environment variable, but I can't find it right know) and see what it says/reports. I bet that "something input related" happens which triggers the bug in Qt. The fix is to upgrade to a fixed Qt version (sorry, I don't know which releases contain the fix) or setting `QT_XCB_NO_XI2_MOUSE=1` in the environment of all Qt-using programs. (= Duplicate of #415)
* Duplicate of #670 (wait, is it *only* `Mod4-Shift-C` for you or do you really see the behavior that is described in #670?)

**Comment by maksbotan** [comment]
Did I get it wrong? http://awesome.naquadah.org/changelogs/v3.5.8
I thought that this changelog meant that keyboard layout issue in wibox is fixed in 3.5.8. If it's not so, sorry. I'll check git master.

Concering Qt: yeah, I've seen that report, but it mentions mouse plugging in the title, so I thought it was only partially relevant. Will try that env var fix.

And for #670. Yes, chrome stops reacting to _any_ input, including text input in fields, `Ctrl-W` and so on. And `Mod3-Shift-C` too. Moreover, it happens not only after opening links from other apps. And I don't have any rules for chrome in my `awesome.rc`.

**Comment by psychon** [comment]
* Commit f128b313130ad10dfa6a fixed a bug where awesome grabbed keys on the wrong window when the keyboard layout changed. This fixes a regression introduced with 3.5.7. AFAIK this has nothing to do with non-English (e.g. German works to and it should even get the different z/y key right). The old problem with some keyboard layouts is (according to the above link to awesome.naquadah.org) about some keyboard layouts where XKB sends weird stuff to non-XKB-enabled client. That link talks about the X server sending things like "Ctrl and Mod2 pressed" in key events even though Ctrl is released and there is no Mod2 key. Awesome then ignores these keys because there is no key binding defined for Ctrl+Mod2+Whatever.

* I will send any bug reports that say "it works again after restarting all Qt applications" to that bug report. This description just sounds too much like "Qt steals our events!". Some duplicate e.g. reported that this issue appeared after suspend/resume, so apparently there are many possible situations where the X11 server "thinks" some hotplugging happened.

* Ok, so this part sounds like it really belongs to #670. As I said, I have no idea about this one, but since it seems to be related to the version number of Chrome and only happens in Chrome... We can talk about this issue there.
